rpc/connection.go

```diff
@@ -608,6 +623,9 @@ type connectionClient struct {
 var _ GenericClient = connectionClient{}
 
 func (c connectionClient) Call(ctx context.Context, s string, args interface{}, res interface{}) error {
+	if isWithFireNow(ctx) || c.conn.initialReconnectBackoffWindow != 0 {
```

Comment:
Shouldn't this be `&&`, not `||`?  (And maybe in reverse order so the cheaper check goes first.)  With `||` it would make any RPC interrupt the wait, and I thought pings weren't supposed to.